app/src/main/java/org/p2p/wallet/auth/web3authsdk/Web3AuthApiClient.kt

```diff
@@ -45,72 +44,77 @@ class Web3AuthApiClient(
         }
     }
 
-    override fun triggerSilentSignUp(
-        socialShare: String,
-        handler: Web3AuthSignUpCallback
-    ) {
-        this.handler = handler
-        onboardingWebView.evaluateJavascript(
-            generateFacade(
-                type = "signup",
-                jsMethodCall = "triggerSilentSignup('$socialShare')"
-            ),
-            null
-        )
+    override suspend fun triggerSilentSignUp(socialShare: String): Web3AuthSignUpResponse {
+        return suspendCancellableCoroutine {
+            this.continuation = it
+
+            onboardingWebView.evaluateJavascript(
```

Comment:
You can move it to func with 2 args evaluateJavascript(type, methodCall)

Reply:
i would rather not to touch it - the code was written by Kostya, and it's very dangerous